notebooks/03-query/09-functions-case.ipynb

```diff
@@ -0,0 +1,499 @@
+    "WHERE session = 'S89403399'\n",
+    "'''\n",
+    "\n",
+    "display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ba25bdec-c11a-4423-bb5c-b38f2d09944b",
+   "metadata": {},
+   "source": [
+    "## CASE functions at query time\n",
+    "\n",
+    "The `CASE` function provides if-then-else behaviour in two forms.\n",
+    "\n",
+    "- The simplest form switches between different outputs based on the value of a dimension.\n",
+    "- The searched form allows for more complex comparison operations.\n",
+    "\n",
+    "This notebook uses the SQL versions of `CASE`. Native (JSON-based) versions [are also available](https://druid.apache.org/docs/27.0.0/querying/math-expr#general-functions), allowing them to be used in streaming ingestion, for example.\n",
+    "\n",
+    "### Simple CASE\n",
+    "\n",
+    "The following SQL shows a simple CASE statement being used to tweak the results of a query based on the browser that someone was using on the site.\n",
+    "\n",
+    "The first parameter for the `CASE` is the dimension, and then each `WHEN` is followed by a specific value to compare with."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "984358e6-4895-49a9-8f32-1baa14c7123d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql='''\n",
+    "SELECT\n",
+    "  browser,\n",
+    "  AVG(\"session_length\") AS \"average_session_length\",\n",
+    "  AVG(\n",
+    "      CASE \"browser\"\n",
+    "          WHEN 'IE' THEN session_length * 2\n",
+    "          WHEN 'Chrome' THEN session_length / 2\n",
+    "          ELSE session_length\n",
+    "          END\n",
+    "          ) AS \"average_session_length_maybe\"\n",
+    "FROM \"example-koalas-conditions\"\n",
+    "WHERE TIME_IN_INTERVAL(__time,'2019-08-25T0/PT1H')\n",
+    "GROUP BY 1\n",
+    "'''\n",
+    "\n",
+    "display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "29bb64c9-9c4c-476b-9718-51c1c35a3d05",
+   "metadata": {},
+   "source": [
+    "### Searched CASE\n",
+    "\n",
+    "The second form of `CASE` is not pinned to a specific dimension or expression, allowing for more complex conditions.\n",
+    "\n",
+    "A common use of `CASE` is to tag particular events as significant.\n",
+    "\n",
+    "With some artistic license, we can apply this to the example dataset. The SQL statement below flags some events as impressions (when someone first saw an image) and others as conversions (when someone completed the task and finished clearing the image)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e8f68a70-e629-4ba7-a28d-5ba282d8469b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql='''\n",
+    "SELECT\n",
+    "  CASE WHEN (\"event_type\" = 'GoodLoad') THEN 'yes' ELSE 'no' END AS \"isImpression\",\n",
+    "  CASE WHEN (\"event_subtype\" IS NULL AND \"event_type\" = 'LayerClear') THEN 'yes' ELSE 'no' END AS \"isConversion\",\n",
```

Comment:
I like this example! Suggest moving `event_type` to the first part of the AND, to keep with the same ordering as the description earlier.

Reply:
Ditto for the below SQL statement

Reply:
Nice. This is much clearer :)